### PR TITLE
testing shortening PR workflow

### PR DESCRIPTION
**Pull request by vsoch**
Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by trws** [comment]
Are these build failures expected? It kinda looks like missing directories?

**Comment by vsoch** [comment]
I think what is happening is that one is generated in the container and the other outside, so they are different - checking now with an echo!

**Comment by vsoch** [comment]
Huh the build matrices shouldn't all be running like that without changed files - let me look into that too.

**Comment by vsoch** [comment]
okay the issue was passing an empty string for parsed files (none found) which defaults to the PWD and will trigger all! I added another hack to fix it lol.

**Review by davidbeckingsale — APPROVED** [review]

File: .github/scripts/combine_results.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+if [[ "${result}" == "" ]]; then
+  result="[]"
+fi
+
+if [[ "${result_scheduled}" == "" ]]; then
+  result_scheduled="[]"
+fi
+
+# Combine results into new output
+result=$(python -c "import json; print(json.loads('${result}') + json.loads('${result_scheduled}'))")
```
Comment (trws) [inline .github/scripts/combine_results.sh:14]: That's a slick way to do that, will have to remember this.

Comment (vsoch) [reply .github/scripts/combine_results.sh:14]: and when you say slick - you mean haaack lol

File: .github/workflows/dockerfiles.yaml [hunk 2]
```diff
@@ -18,94 +18,99 @@ on:
...
+          flags: "--no-empty-build-args"
+      # END PULL REQUEST / PUSH checkout and file derivation =======================
+
+      # START SCHEDULED checkout and file derivation =======================
       - uses: actions/checkout@v2
-        if: github.event_name != 'pull_request'
+        if: github.event_name == 'schedule'
         with:
            fetch-depth: 0
+           ref: ${{ github.event.pull_request.head.ref }}
```
Comment (trws) [inline .github/workflows/dockerfiles.yaml:70]: Does this mean fetch the pull request head on scheduled runs?

Comment (vsoch) [reply .github/workflows/dockerfiles.yaml:70]: oh that should probably be removed. It was there for when uptodate was trying to manage the git history.

Comment (vsoch) [reply .github/workflows/dockerfiles.yaml]: Removed.

File: .github/scripts/combine_results.sh (hunk 1 above)
Comment (trws) [reply .github/scripts/combine_results.sh:14]: LoL, I mean, can't it be both? 😁
If jq isn't there, that's a great low-dependency way to do it, that's more reliable than anything else I can think of for handling a hunk of json on a minimal linux.

**Review by trws — APPROVED** [review]

**Pull request merged**